src/wasm/wasm-stack.cpp

```diff
@@ -2500,7 +2500,10 @@ void BinaryInstWriter::emitMemoryAccess(size_t alignment,
   if (memoryIdx > 0) {
     o << U32LEB(memoryIdx);
   }
-  o << U32LEB(offset);
+  if (memory64)
+    o << U64LEB(offset);
+  else
+    o << U32LEB(offset);
```

Comment:
I don't think binaryen allows if/else blocks without braces does it?

Reply:
address comment, thanks